symbolic-unwind/src/evaluator/mod.rs

```diff
@@ -95,48 +105,60 @@ impl<'memory, A, E> Evaluator<'memory, A, E> {
         self
     }
 
-    /// Sets the evaluator's constant map to the given map.
-    pub fn constants(mut self, constants: BTreeMap<Constant, A>) -> Self {
-        self.constants = constants;
+    /// Sets the evaluator's register map to the given map.
+    pub fn registers(mut self, registers: BTreeMap<Register, A>) -> Self {
+        self.registers = registers;
         self
     }
 
-    /// Sets the evaluator's variable map to the given map.
-    pub fn variables(mut self, variables: BTreeMap<Variable, A>) -> Self {
-        self.variables = variables;
-        self
+    /// Add a new register rule to the evaluator. These rules are used by
+    /// [`evaluate_register`](Self::evaluate_register)
+    /// and [`evaluate_all_registers`](Self::evaluate_all_registers).
+    pub fn add_cfi_rule(&mut self, register: Register, expr: Expr<A>) {
+        if register == Register::cfa() {
```

Comment:
`Register::cfa()` actually allocates, and we use it in a couple of places just for these comparisons.

Reply:
Yeah, I know :/
I was kinda working under the assumption that at some point we are going to make `Register` non-allocating and the problem solves itself. That being said, I could also put an `is_cfa` method on register that performs this test in the interim.